src/components/tables/achievements-table.tsx

```diff
@@ -7,34 +7,126 @@ import {
   TableRow,
   Typography,
 } from "@mui/material";
-
-import { type LoggedUserProfileData } from "~/types/index.js";
+import {
+  type AchievementId,
+  type LoggedUserProfileData,
+} from "~/types/index.js";
+import { api } from "~/utils/api.js";
 import { Header } from "./components/index.js";
 import { text } from "~/assets/text/index.js";
+import { matchAchievements } from "~/server/service/achievements.js";
+import { POINTS_ACHIEVEMENTS } from "~/constants/main.js";
 import { styles } from "./styles.js";
 
 interface Props {
   playerProfile: LoggedUserProfileData;
 }
 
-//TODO: Add the achievements to the player profile
+const COLUMN_WIDTH = { sm: "5.0%", md: "20.5%", lg: "30%" };
+
+type AchievementRow = {
+  name: string;
+  description: string;
+  streak: string;
+  pointsEarned: number;
+  pointValue: number;
+};
+
 export const AchievementsTable: FC<Props> = ({ playerProfile }) => {
+  const achievements = api.user.getUserAchievements.useQuery(undefined, {
+    retry: false,
+  });
+
   if (!playerProfile) return;
 
+  const achievementsRecord: Record<AchievementId, string> = {
+    busterStreak: "x5",
+    fiveDayStreak: "x2",
+    firstTimer: "0",
+    beginnersLuck: "0",
+    goodBust: "0",
+    masterBuster: "0",
+    realHuman: "0",
+    dailyStreakCounter: "0",
+  };
+
+  const countAchievement = (achievementId: AchievementId) => {
```

Comment:
So this is actually computing achievement points. Let's call it `computePoints`.